subprojects/logging/src/main/java/org/gradle/internal/logging/sink/OutputEventRenderer.java

```diff
@@ -329,6 +352,10 @@ public void onOutput(OutputEvent event) {
                 return;
             }
             this.maxWorkerCount.set(newMaxWorkerCount);
+        } else if (event instanceof EndOutputEvent) {
+            if (updateNowExecutor != null) {
+                updateNowExecutor.shutdownNow();
```

Comment:
According to the documentation, the `shutdown()` method will wait the remaining time (for the scheduled) and execute the task before completely shutting down. The `shutdownNow()` will simply cancel any task waiting to be executed (won't wait) and will try to interrupt the currently executing task (if we check for interruption). If the interruption isn't handled, it will wait for the task to finish its execution. In our case, it's alright to wait for the end of the execution of the current task but would be bad, performance wise, to wait for the next scheduling of the task.